--- include/string.h
+++ include/string.h
@@ -8,6 +8,8 @@
 //
 //===----------------------------------------------------------------------===//
 
+#include_next <string.h>
+
 #ifndef _LIBCPP_STRING_H
 #define _LIBCPP_STRING_H
 
--- include/wchar.h
+++ include/wchar.h
@@ -8,6 +8,8 @@
 //
 //===----------------------------------------------------------------------===//
 
+#include_next <wchar.h>
+
 #if defined(__need_wint_t) || defined(__need_mbstate_t)
 
 #if !defined(_LIBCPP_HAS_NO_PRAGMA_SYSTEM_HEADER)
